client/src/js/components/main/updatePasswordForm.jsx

```diff
@@ -1,130 +1,195 @@
-// import { connect } from "react-redux";
-import Api from "../../utils/api";
-import { Welcome } from "./../partials/";
+// import Api from '../../utils/api';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import {
+  updatePassword
+} from '../../actions/authAction';
+import { Welcome } from './../partials/';
 import '../../../styles/index.scss';
 
-class UpdatePassword extends React.Component {
+/**
+ * @typedef {object} event
+ */
+
+/**
+ * Display UpdatePasswordForm
+ * @class UpdatePasswordForm
+ * @extends {React.Component}
+ * @param {any} props
+ */
+class UpdatePasswordForm extends React.Component {
+  /**
+   * Creates an instance of UpdatePasswordForm
+   * @param {any} props
+   * @memberof UpdatePasswordForm
+   * @return {void}
+   */
   constructor(props) {
     super(props);
     this.onResetPassword = this.onResetPassword.bind(this);
     this.onFocus = this.onFocus.bind(this);
     this.state = {
-      status: "",
-      message: "",
-      updateButtonClassName: "btn waves-effect waves-light"
+      status: '',
+      message: '',
+      updateButtonClassName: 'btn waves-effect waves-light'
     };
   }
 
-  onChange(e) {
+  /**
+   * onChange Method
+   * @param {event} event
+   * @return {void} void
+   */
+  onChange(event) {
     this.setState({
-        [e.target.name]: e.target.value
+      [event.target.name]: event.target.value
     });
   }
 
+  /**
+   * onFocus Method
+   * @param {void} empty
+   * @return {void}
+   */
   onFocus() {
     this.setState({
-      status: "",
-      message: ""
+      status: '',
+      message: ''
     });
   }
 
-  onResetPassword(e) {
-    e.preventDefault();
-    if (this.state.status === "success") {
+  /**
+   * onResetPassword
+   * @param {event} event
+   * @return {void} null
+   */
+  onResetPassword(event) {
+    event.preventDefault();
+    if (this.state.status === 'success') {
       this.setState({
-        status: "",
-        message: "",
-        updateButtonClassName: "btn waves-effect waves-light disabled"
+        status: '',
+        message: '',
+        updateButtonClassName: 'btn waves-effect waves-light disabled'
       });
       return null;
     }
-    if (this.password.value !== "") {
-      const hash = window.location.href.split("/")[5];
-    //   console.log(hash);
+    if (this.password.value !== '') {
+      const hash = window.location.href.split('/')[5];
       if (hash === undefined) {
-        window.location = "/#/login";
+        window.location = '/#/login';
+        // redirecting
+        // this.props.history.push('/login');
         return;
       }
-      const password = `password=${this.password.value}`;
-      console.log(password);
-      Api(password, `/api/v1/users/reset/${hash}/`, "POST", null).then(
-        (response) => {
+      this.props.updatePassword(this.password.value, hash)
+        .then((response) => {
           if (response.data.error === undefined) {
```

Comment:
i think this should work `if (!response.data.error)` rather than specifying `undefined`